plugin.video.telebasel/resources/lib/telebasel.py

```diff
@@ -0,0 +1,537 @@
+        """
+        log('build_main_menu')
+        main_menu_list = [
+            {
+                # All shows
+                'name': LANGUAGE(30050),
+                'mode': 10,
+                'isFolder': True,
+                'displayItem': True,
+            }, {
+                # Newest shows
+                'name': LANGUAGE(30051),
+                'mode': 11,
+                'isFolder': True,
+                'displayItem': True,
+            }, {
+                # Live TV
+                'name': LANGUAGE(30072),
+                'mode': 12,
+                'isFolder': True,
+                'displayItem': True,
+            },
+        ]
+        for menu_item in main_menu_list:
+            if menu_item['displayItem']:
+                list_item = xbmcgui.ListItem(menu_item['name'])
+                list_item.setProperty('IsPlayable', 'false')
+                list_item.setArt({'thumb': ICON})
+                purl = self.build_url(
+                    mode=menu_item['mode'], name=menu_item['name'])
+                xbmcplugin.addDirectoryItem(
+                    handle=int(sys.argv[1]), url=purl,
+                    listitem=list_item, isFolder=menu_item['isFolder'])
+
+    def retrieve_shows(self, extract_channel_id=False):
+        """
+        Retrieve basic infos about the available shows from the website.
+        It returns a list of dictionaries with the keys
+        'title', 'page', 'image' and optionally 'channel_id'.
+
+        Keyword arguments:
+        extract_channel_id  -- Extract the channel ids (default: False)
+        """
+        log('retrieve_shows, extract_channel_id=%s' % extract_channel_id)
+        url = HOST_URL + '/mediathek'
+        url_result = self.open_url(url, use_cache=True)
+        soup = BeautifulSoup(url_result, 'html.parser')
+        shows_soup = soup.find_all('a', {'class': 'tb-a-unstyled'})
+
+        shows = []
+        image_regex = r'background-image\s*:\s*url\(\'(?P<url>.+)\''
+        for ssoup in shows_soup:
+            page = ssoup.attrs['href']
+            m = re.search(image_regex, str(ssoup))
+            image_url = m.group('url')
+
+            filename = image_url.split('/')[-1]
+            filename_regex = r'(?P<trunk>.+)-\d+x\d+\.(?P<ext>.+)'
+            match = re.match(filename_regex, filename)
+            if match:
+                trunk = match.group('trunk')
+                ext = match.group('ext')
+                image_url = '/'.join(
+                    image_url.split('/')[:-1]) + '/' + trunk + '.' + ext
+
+            title = ssoup.h2.text
+            # The element 'Telebasel Archiv' refers to the old
+            # website. We need to skip this:
+            if 'Archiv' in title:
+                continue
+
+            shows.append({'title': title, 'page': page, 'image': image_url})
+
+        if extract_channel_id:
+            feed_urls = [x['page'] + '&podcast' for x in shows]
+            pool = ThreadPool(len(feed_urls))
```

Comment:
last time I checked multiprocessing is prone to errors in some platforms (don't remember exactly which), maybe its solved by now. In how many different platforms have you tested your addon?
Normally devs try to implement a fallback in case multiprocessing fails, see: https://github.com/xbmc/repo-plugins/blob/krypton/plugin.video.locast/resources/lib/locast.py

Maybe @Lunatixz can explain a bit better where it is prone to fail

Reply:
I believe this issue is limited to Kodi variants, ie Openelec, LibreELEC which ship lightweight python packages. Please include a fallback to Threadpool for incompatible systems.

Reply:
Sorry, I did not see your reply. How do I know how to trigger the fallback? Will these distributions with lightweight python variants not include the `multiprocessing` module, so `import multiprocessing.dummy` will throw an `ImportError`? Or how do I know when to use the fallback?